src/state.rs

```diff
@@ -73,11 +105,48 @@ impl Default for Tx {
     }
 }
 
-// impl fmt::Debug for Coin {
-//     fn fmt(&self, f: &mut Formatter<'_>) -> fmt::Result {
-//         format!("{} {}", ConstLenStr(self.denom.clone()), ConstLenStr(self.amount.to_string())).fmt(f)
-//     }
-// }
+pub fn store_swap<S: Storage>(
+    store: &mut S,
+    destination: String,
+    amount: Uint128
+) -> StdResult<()> {
+    let mut store = PrefixedStorage::new(PREFIX_SWAP, store);
+    let mut store = AppendStoreMut::attach_or_create(&mut store)?;
+
+    let nonce = store.len();
+    let swap = Swap{
+        destination,
+        amount,
+        nonce
+    };
+    store.push(&swap)
+}
+
+pub fn get_swap<A: Api, S: ReadonlyStorage>(
+    api: &A,
+    storage: &S,
+    nonce: u32,
+) -> StdResult<Swap> {
+    let mut store = ReadonlyPrefixedStorage::new(PREFIX_SWAP, storage);
+
+    // Try to access the storage of txs for the account.
+    // If it doesn't exist yet, return an empty list of transfers.
+    let store = if let Some(result) = AppendStore::<Swap, _>::attach(&store) {
+        result?
+    } else {
+        return Err(StdError::generic_err("Tx does not exist"));
+    };
```

Comment:
As the comment implies, I don't think you want to return an error here, and this error message feels incorrect to me anyways.
look at what i do in this case under `get_transfers`. If `AppendStore::attach` returns `None` that just means you havent written anything to it yet, since you maybe called the query endpoint before making any swaps.